hatchet/readers/hpctoolkit_reader_latest.py

```diff
@@ -384,49 +665,117 @@ def _read_cct(
+        return GraphFrame(
+            Graph(self._graph_roots),
+            dataframe,
+            inc_metrics=inc_metrics,
+            exc_metrics=exc_metrics,
+            default_metric=self._time_metric,
+            metadata=metadata,
+        )
 
-            for im in list(self._inclusive_metrics.values()):
-                if im in table.columns.tolist():
-                    inclusive_metrics.append(im)
+    def _read_summary_profile(
+        self,
+    ) -> None:
+
+        with open(self._profile_file, "rb") as file:
+            file.seek(FILE_HEADER_OFFSET)
+            formatProfileInfos = "<QQ"
+            profile_db = file.read(struct.calcsize(formatProfileInfos))
+            (szProfileInfos, pProfileInfos) = safe_unpack(
+                formatProfileInfos, profile_db, 0
+            )
 
-            for em in (list(self._exclusive_metrics.values()),):
-                if em in table.columns.tolist():
-                    exclusive_metrics.append(em)
+            file.seek(pProfileInfos)
+            profile_db = file.read(szProfileInfos)
+            (pProfiles,) = safe_unpack("<Q", profile_db, 0)
 
-            graphframe = GraphFrame(
-                Graph(self._cct_roots),
-                table,
-                inc_metrics=inclusive_metrics,
-                exc_metrics=exclusive_metrics,
-                default_metric=self._time_metric,
+            (nValues, pValues, nCtxs, _, pCtxIndices) = safe_unpack(
+                "<QQLLQ", profile_db, pProfiles - pProfileInfos
             )
 
-            print("DATA IMPORTED")
-            return graphframe
+            file.seek(pCtxIndices)
+            formatCtxs = "<LQ"
+            cct_sub_db = file.read(nCtxs * struct.calcsize(formatCtxs))
+
+            file.seek(pValues)
+            formatValues = "<Hd"
+            values_sub_db = file.read(nValues * struct.calcsize(formatValues))
+
+            for i in range(nCtxs):
+                (ctxId, startIndex) = safe_unpack("<LQ", cct_sub_db, 0, i)
+                cct_id = ctxId
+
+                if i == nCtxs - 1:
+                    end_index = nValues
+                else:
+                    (_, end_index) = safe_unpack(formatCtxs, cct_sub_db, 0, i + 1)
+
+                self._summary_profile[cct_id] = {}
+
+                for j in range(startIndex, end_index):
+                    (metricId, value) = safe_unpack("<Hd", values_sub_db, 0, j)
+
+                    if metricId in self._metric_descriptions:
+                        self._summary_profile[cct_id][
+                            self._metric_descriptions[metricId]
+                        ] = value
 
     def read(self) -> GraphFrame:
         self._read_metric_descriptions()
-        self._read_summary_profile()
+        if (
+            not self._parallel_profiles_mode
+            or self._min_application_percentage_time is not None
+        ):
+            self._read_summary_profile()
+
+        if self._parallel_profiles_mode:
```

Comment:
@dyokelson the case where both `self._parallel_profiles_mode == True` and `self._min_application_percentage_time is not None` should be allowed. In that case, although I'm reading parallel profiles, I still need to extract the summary one because of the `self._min_application_percentage_time` condition. When this flag is on, we check whether the execution time of a node is greater or equal to some threshold. This check is always performed on the data stored in the summary profile (even when fetching parallel profiles). When fetching parallel profiles, we will use time in the summary profile to check whether nodes should be removed from the tree or not, and after this condition is checked, we extract data for each parallel profile for that node (if we decide to keep them in the tree).